overlay_test.go

```diff
@@ -25,10 +25,7 @@ func (po *ProtocolOverlay) Start() error {
 
 func (po *ProtocolOverlay) Dispatch() error {
 	if po.failDispatch {
-		go func() {
-			// hopefully we write to channel after `Dispatch` returns
-			po.failChan <- true
-		}()
+		po.failChan <- true
```

Comment:
could you add a comment here to stay ideally this should happen after `return errors.New("Dispatch failed")`? otherwise, if the runtime decides to execute in the following order:
```
1, po.failChain <- true
2, assert.Contains(t, log.GetStdOut()+log.GetStdErr(), "Dispatch failed")
3, return errors.New("Dispatch failed")
```
the test would fail (although i have no idea if it's possible for the runtime to do something like that)